File: apps/java-app/src/main/java/com/example/Greeter.java
package com.example;

public class Greeter {
    private final String name;

    public Greeter(String name) {
        this.name = name;
    }

    public String greet() {
        return "Hello from Java, " + name + "!";
    }

    public static void main(String[] args) {
        String target = args.length > 0 ? args[0] : "Bazel";
        System.out.println(new Greeter(target).greet());
    }
}

File: apps/cpp-lib/greeter.cc
#include "apps/cpp-lib/greeter.h"
#include "libs/common/version.h"

namespace example {

Greeter::Greeter(std::string name) : name_(std::move(name)) {}

std::string Greeter::Greet() const {
    return "Hello from C++, " + name_ + "! (" + common::GetBuildInfo() + ")";
}

}  // namespace example
